singleton + relative methods + event args + fix pinch to scale + mark events as attended

arturoc opened the pull request:
have been taking a look at this. it works great and using it in combination with ofNode, makes it super easy to get pinch gestures working, i'll try to add an example. i've added some things and fixed some issues:

singleton: in case you want to use the addon from different classes
avoids to pass a reference around, and it's more similar to how other
OF event classes work

relative methods: when doing a pinch gesture it's easier to transform
based on the relative position from the last event, that way you don't
need the transformations at the beginning of the pinch gesture, only the
last ones or do relative transformations.

event args: added event args (that internally call the corresponding
ofxGesture methods) to make the syntax less verbose and more uniform
with how other OF events work

fix pinch to scale: was reversed, scaling up when closing the fingers
instead of the opposite

mark events as attended: now the events propagate the return value of
the listeners which make it possible to stop the touch events from being 
propagated further if a listener attended a pinch or pan event already

arturoc committed:
singleton + relative methods + event args + fix pinch to scale

singleton: in case you want to use the addon from different classes
avoids to pass a reference around, and it's more similar to how other
OF event classes work

relative methods: when doing a pinch gesture it's easier to transform
based on the relative position from the last event, that way you don't
need the transformations at the beginning of the pinch gesture, only the
last ones or do relative transformations.

event args: added event args (that internally call the corresponding
ofxGesture methods) to make the syntax less verbose and more uniform
with how other OF events work

fix pinch to scale: was reversed, scaling up when closing the fingers
instead of the opposite

tallavi commented:
Very nice! I'll integrate this version in my project.

BTW, I saw commits on OF's master branch, with additions you made in OFAndroid to support gestures. Aren't they a little redundant when there's a way to do gestures cross platform?

arturoc commented:
yes they are, i added them yesterday because i needed pinch but android doesn't have rotate, only scale.

there's still somethings in this addon that are slightly problematic. for example the gestures on the native android implementation are only triggered after some movement happens not right after putting 1 or 2 fingers on the screen.

i think i'm going to remove the scale gestures i added yesterday and we can keep developing this addon and try to merge it for the next major release

tallavi commented:
All these issues can be handled and more. We can have a tolerance value telling us how many pixels (or screen percentages) needs to be moved before a gesture is declared. 

Also, we can handle more gestures: double tap, triple tap, swipe (with built in momentum and friction values managed by ofxGestures) and more.. 

In short: every OS specific crazyness can be modeled and implemented here, and be accessed by all touch enabled OSs.
